tests/has/devfile_source.go

```diff
@@ -103,6 +111,108 @@ var _ = framework.HASSuiteDescribe("devfile source", func() {
 		Expect(err).NotTo(HaveOccurred())
 		Expect(component.Name).To(Equal(QuarkusComponentName))
 	})
+
+	It("Wait for component pipeline to be completed", func() {
+		err := wait.PollImmediate(20*time.Second, 10*time.Minute, func() (done bool, err error) {
+			pipelineRun, _ := framework.HasController.GetComponentPipeline(QuarkusComponentName, RedHatAppStudioApplicationName, AppStudioE2EApplicationsNamespace)
+
+			for _, condition := range pipelineRun.Status.Conditions {
+				klog.Infof("PipelineRun %s reason: %s", pipelineRun.Name, condition.Reason)
+
+				if condition.Reason == "Failed" {
+					return false, fmt.Errorf("Component %s pipeline failed", pipelineRun.Name)
+				}
+
+				if condition.Status == corev1.ConditionTrue {
+					return true, nil
+				}
+			}
+			return false, nil
+		})
+		Expect(err).NotTo(HaveOccurred(), "Failed component pipeline %v", err)
+	})
+
+	It("Create GitOps Deployment", func() {
+		gitOpsRepository := ObtainGitOpsRepositoryUrl(application.Status.Devfile)
+		gitOpsRepositoryPath := fmt.Sprintf("components/%s/base", QuarkusComponentName)
+
+		deployment, err := framework.GitOpsController.CreateGitOpsCR(GitOpsDeploymentName, AppStudioE2EApplicationsNamespace, gitOpsRepository, gitOpsRepositoryPath, GitOpsRepositoryRevision)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(deployment.Name).To(Equal(GitOpsDeploymentName))
+	})
+
+	It("Check component deployment health", func() {
+		Eventually(func() bool {
+			deployment, _ := framework.HasController.GetComponentDeployment(QuarkusComponentName, AppStudioE2EApplicationsNamespace)
+			if deployment.Status.AvailableReplicas == 1 {
+				klog.Infof("Deployment %s is ready", deployment.Name)
+				return true
+			}
+
+			return false
+		}, 3*time.Minute, 10*time.Second).Should(BeTrue(), "Component deployment didn't become ready")
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	It("Check component service health", func() {
+		service, err := framework.HasController.GetComponentService(QuarkusComponentName, AppStudioE2EApplicationsNamespace)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(service.Name).NotTo(BeEmpty())
+		klog.Infof("Service %s is ready", service.Name)
+	})
+
+	It("Verify component route health", func() {
+		route, err := framework.HasController.GetComponentRoute(QuarkusComponentName, AppStudioE2EApplicationsNamespace)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(route.Spec.Host).To(Not(BeEmpty()))
+		klog.Infof("Component route host: %s", route.Spec.Host)
+	})
```

Comment:
I see that this piece of code was removed in [this PR](https://github.com/redhat-appstudio/e2e-tests/pull/62/files#diff-7b20c4c262d6271985633ce1d9b3957a82a18f4ebf48e1da66fcf35624e68a1aL107-L133)

Should we remove it from here too?

Reply:
+1

Reply:
Removed it. Thanks!